crates/p2p/src/client/peer_agnostic.rs

```diff
@@ -332,6 +348,71 @@ impl Client {
     }
 }
 
+/// Does not block the current thread.
+fn compute_contract_addresses(
+    deploy_account: HashMap<BlockHash, Vec<super::types::RawDeployAccountTransaction>>,
+    tx: tokio::sync::oneshot::Sender<Vec<(BlockHash, Vec<TransactionVariant>)>>,
+) {
```

Comment:
Several thoughts; feel free to disregard them all.

- Would it be better to have a `pathfinder_common::DeployTransaction::compute_contract_address()`? This would also mean spending less time working with the dto type. (I haven't actually checked the usage so maybe it makes sense to not parse into common types at all).
- Let the caller make the decision for rayon or not, since its just a single wrapper? Or is this used in many locations?
- I think the oneshot channel is basically what `tokio::task::spawn_blocking` would do?

Reply:
1. this is a function for the one particular purpose/use so I didn't want to litter the caller's block of code with rayon
2. imo `tokio::task::spawn_blocking` has some disadvantages in this situation:
- we're using one more thread which is needlessly waiting for rayon to finish, and this thread comes from tokio's "blocking task thread pool", which is also utilized by db access
- you still need to communicate the result from rayon to tokio's "blocking task" pool of threads which means using
    - an (std) channel anyway to send the result from rayon to tokio's "blocking task pool" thread
    - using a rayon::scope which means the same as above

  and in both cases it is just more code.
  
Generally I think that using `tokio::task::spawn_blocking` or `rayon::scope` in such situations should be the last resort when there are many computations referring to `rayon` in various stages like for example when verifying blocks and you need some "root" thread to begin with.

Reply:
Sorry I missed a step - you don't need `rayon::spawn` because using `par_iter` already does this channel business for you I think?

So in my mind you could:

- make this function `async`
- use `spawn_blocking` + `par_iter` to invoke rayon
- I think we're just replicating the tokio blocking thread pool with the manual channel

I'm like... only 40% certain 🙃 So let's move on if you're happy as is 👍

Reply:
Ah! I forgot the `par_iter` part :D, yeah this :arrow_up: makes sense!